department-of-veterans-affairs/va.gov-cms
# VACMS-15506 aging content fwb mvp with expirable content module

## Description
This PR:

- Adds the 'expirable_content' module from contrib which allows any content entity type to have a calculated expiration date.
- Adds a new message template for Full Width Banner warning notifications.
- Adds new ECA model to trigger the FWB warning notifications
- ECA model changes to correct processing logic flaw
- Adds relevant changes for theming the new template (theme hook, mime configuration, etc)
- Adds a patch to the token module to support computed fields
- Changes the Aging Content View to use the new fields and filters provided by the Expirable Content module.
- Adds a deploy hook to seed the initial expiration dates for FWB nodes.
- A ddev tweak to allow debugging from the command line in PHPStorm

Relates to #15506 

## Testing done
Tested locally using the provided script to generate FWB nodes, and send the notification emails by running cron both in the UI and from Drush.

## Screenshots

![Screenshot 2024-04-25 at 12 45 10 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/a54ad7c3-f9c0-48b2-960a-4252bec45330)
![Screenshot 2024-04-25 at 12 41 58 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/d0a57733-999b-4ae3-97ba-1b8a76c75a09)
![Screenshot 2024-04-25 at 12 40 40 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/cf0db6c0-c68e-4956-ab8a-510b0af4c538)
![Screenshot 2024-04-25 at 12 39 37 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/5eff9578-7850-45fe-8095-c5f9855dfef9)

## QA steps

As a Tugboat user and Drupal admin:
1. Visit the Tugboat dashboard
2. Click the 'Terminal' link for the PHP service
3. In the window that opens, run the following: `drush scr scripts/content/VACMS-15506-generate-test-nodes.php`.
4. Visit the [Aging Content](https://pr17732-ubppbsl9qxkfbbjvvag7dlejqvwlbpts.ci.cms.va.gov/admin/structure/views/view/aging_content) View
   - [x] In the preview section, validate you see the node with title 'Expirable Content Test Node:expired'. **note you may see other nodes there and that is ok**
5. Visit the [Warning](https://pr17732-ubppbsl9qxkfbbjvvag7dlejqvwlbpts.ci.cms.va.gov/admin/structure/views/view/aging_content/edit/fwb_warn) display
   - [x] In the preview section, validate that you see a node with title 'Expirable Content Test Node:warn'. **note you may see other nodes there and that is ok**
6. Visit the [ECA model](https://pr17732-ubppbsl9qxkfbbjvvag7dlejqvwlbpts.ci.cms.va.gov/admin/config/workflow/eca) landing page.
7. Enable the first model 'Aging Content: Warn FWB' by clicking the Operations dropdown and selecting 'enable'. You should notice the page reloads automatically as the action is done via ajax.
8. Run CRON by hovering over the VA icon in the toolbar menu and selecting 'cron'. Wait for the page to reload.
9. Run CRON a second time. (the first time queued the message, the second time sends it)
10. Visit the Tugboat [dashboard](https://tugboat.vfs.va.gov/660ebb84b4e1649ad17fe163)
   - [x] Validate that under the 'Captured Mail' there exists an email with the title 'Please update or archive your Full-width banner by 2024-04-24'
11. Visit the [ECA model](https://pr17732-ubppbsl9qxkfbbjvvag7dlejqvwlbpts.ci.cms.va.gov/admin/config/workflow/eca) landing page.
12. Disable the 'Aging Content: Warn FWB' model and enable the 'Aging Content: Expire FWB' model.
13. Run CRON
14. Run CRON again
15. Visit the Tugboat [dashboard](https://tugboat.vfs.va.gov/660ebb84b4e1649ad17fe163)
   - [x]  Validate that under 'Captured Mail' there exist an email with the title 'Your Full-width banner has displayed for seven (7) days'

### Definition of Done


### Select Team for PR review
